### PR TITLE
Feature: Golden Fish Timer

### PR DESCRIPTION
**Pull request by ItsEmpa**
## What
Adds a Golden Fish Timer, with info on how long it has been since the last rod, time until you can spawn one, etc.
Also adds options to, for example, show certain info as a nametag on top of the golden fish, highlight it when its ready and give a warning when you are close to the time limit of throwing rod.

<details>
<summary>Images</summary>

image
image

</details>

## Changelog New Features
+ Added Golden Fish Timer. - Empa
    * Includes an option to warn you when to throw your rod.
    * Shows how weak the golden fish is, as a nametag.
    * Also works on Stranded.

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/fishing/trophy/GoldenFishTimer.kt [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+import net.minecraft.entity.EntityLivingBase
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+
+class GoldenFishTimer {
+
+    private val config get() = SkyHanniMod.feature.fishing.trophyFishing.goldenFishTimer
+
+    private val group = RepoPattern.group("fishing.goldenfish")
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/fishing/trophy/GoldenFishTimer.kt]: should probably be called patternGroup for consistency with the rest of the mods code

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
code looks good just this one thing

**Review by CalMWolfs — APPROVED** [review]

**Comment by hannibal002** [comment]
idk what i am doing wrong. either the 3 minutes delay are wrong. or you need to fish something, not throw the rod in. 
image

**Comment by ItsEmpa** [comment]
i think the most likely reason is that the detection for wheter or not something actually counts as throwing rod is really weird

**Comment by hannibal002** [comment]
oje
image

**Review by hannibal002 — CHANGES_REQUESTED** [review]
Does not work for me.

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**